src/features/cart/cartModule.ts

```diff
@@ -0,0 +1,208 @@
+export {};
+
+interface CartItem {
+  id: number;
+  title: string;
+  price: number;
+  quantity: number;
+  discountApplied: string[];
+}
+
+interface DiscountCode {
+  code: string;
+  type: "cart" | "item";
+  value: number;
+  itemId?: number;
+}
+
+interface Cart {
+  items: CartItem[];
+  discount: string[];
+}
+
+const discountCodes: Array<DiscountCode> = [
+  {
+    code: "halfDiscount",
+    value: 50,
+    type: "cart",
+  },
+  {
+    code: "fullDiscount",
+    value: 100,
+    type: "cart",
+  },
+  {
+    code: "tfiveDiscount",
+    value: 25,
+    type: "item",
+    itemId: 3,
+  },
+];
+
+const cart: Cart = {
+  items: [],
+  discount: []
+};
+
+const addItem = (product: CartItem): void => {
+  const isExist = cart.items.find(item => item.id === product.id);
+  if(isExist){
+    isExist.quantity++;
+    console.log("Quantity increamented for this product", isExist);
+  } else {
+    cart.items.push(product);
+    console.log(
+      "New Product Added",
+      cart.items.find((item) => item.id === product.id)
+    );
+  }
+};
+
+addItem({
+  id: 1,
+  title: "Item 1",
+  price: 120,
+  quantity: 1,
+  discountApplied: [],
+});
+addItem({
+  id: 2,
+  title: "Item 2",
+  price: 120,
+  quantity: 1,
+  discountApplied: [],
+});
+addItem({
+  id: 3,
+  title: "Item 3",
+  price: 120,
+  quantity: 1,
+  discountApplied: [],
+});
+addItem({
+  id: 4,
+  title: "Item 4",
+  price: 120,
+  quantity: 1,
+  discountApplied: [],
+});
+addItem({
+  id: 5,
+  title: "Item 5",
+  price: 120,
+  quantity: 0,
+  discountApplied: [],
+});
+
+
+
+const removeItem = (id: number): void => {
+  const isExist = cart.items.find(item => item.id === id && item.quantity >= 1)
+  if(isExist) {
+    isExist.quantity = 0;
```

Comment:
This should decrease the quantity by 1. Also if the quantity gets to zero, remove the item from the array.